### PR TITLE
fix(levm): handle expects

### LINKED ISSUE #1074: LEVM: add `#[forbid(clippy::unwrap_used)]`, `#[forbid(clippy::expect_used)]`, and `#[forbid(clippy::panic)]`
1. Add the lints `#[forbid(clippy::unwrap_used)]`, `#[forbid(clippy::expect_used)]`, and `#[forbid(clippy::panic)]` to the `levm` module.
2. Handle the errors or propagate them as needed.

### PR DESCRIPTION
**Pull request by JereSalo**
**Motivation**

**Description**

Partially Closes #1074

### CODE REVIEW

File: crates/vm/levm/clippy_delete.txt
Comment (ilitteri) [inline crates/vm/levm/clippy_delete.txt:1]: Remove this file.

File: crates/vm/levm/src/db.rs [hunk 1]
```diff
@@ -93,12 +93,13 @@ impl Cache {
         self.accounts.insert(*address, account.clone());
     }
 
-    pub fn write_account_storage(&mut self, address: &Address, key: H256, slot: StorageSlot) {
+    pub fn write_account_storage(&mut self, address: &Address, key: H256, slot: StorageSlot) -> Result<(), VMError>{
         self.accounts
             .get_mut(address)
-            .expect("Account should have been cached")
+            .ok_or(VMError::AccountShouldHaveBeenCached)?
```
Comment (ilitteri) [inline crates/vm/levm/src/db.rs:99]: This should be an internal error.

**Review by ilitteri — CHANGES_REQUESTED** [review]
The PR looks good overall. Left one comment.

File: crates/vm/levm/src/db.rs (hunk 1 above)
Comment (JereSalo) [reply crates/vm/levm/src/db.rs]: Done!

File: crates/vm/levm/src/vm.rs
Comment (ilitteri) [inline crates/vm/levm/src/vm.rs:1]: The changes to this file doesn't belong to this PR.

Comment (JereSalo) [reply crates/vm/levm/src/vm.rs:1]: I didn't realize those changes were reverted in `levm_fixes`, I will change it.

Comment (JereSalo) [reply crates/vm/levm/src/vm.rs:1]: I think I solved it with [this revert](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1109/commits/7b2248f4ef8796fc3fe645c0425b88899357d14f). But tell me if you think there is something else that don't belong here.

**Review by ilitteri — APPROVED** [review]
LGTM

**Pull request merged**